### ISSUE TITLE
Home Assistant OS loses network connectivity when Ethernet goes down (router reboot)

### ISSUE THREAD

**Issue by budius** [home-assistant/operating-system#1902]
### Describe the issue you are experiencing

TL;DR:

Home Assistant OS completely loses connectivity if the router reboots.

Details:

My router is "fairly modern" TP-Link Archer AX-50 router and when I setup it a year ago there was this option to nightly reboot, which I selected because I'm a developer and I know memory leak is a real thing and why not let it reboot every night at 4AM.

I never had problems with the nightly reboot. Chromecast, PowerLine network, different raspberry PIs (mini-DLNA, piHole, camera) that I have around, everything would just work. It would reboot and in the morning I wouldn't have known because everything just works.

A month or so ago I got started with Home Assistant, so it's currently a small setup, just to get start and learn configurations with time. rPi 3 that I had already, ordered a Conbee II USB stick and bought some bulbs, sockets and remote on Ikea.

Got the latest 64-bit OS installed on a SansDisk A1 card I had laying around and setup went all fine with 'zha' without issues. But every night I would lose access to Home Assistant. Cannot connect via the Android app, or via homeassistant.local:8123 or via 192,168.2.99:8123, cannot ping from my laptop terminal and cannot see it in the router control panel. The only way to get it back was pulling the plug and turn on again.

Very important to note that Home Assistant and 'zha' was still working as I could click on the remote control all the controls for the lights and sockets were still operational. It is just the Ethernet access that was down.

I tried: a new power supply; a new SD-card; re-install with the 32-bit OS, revert back to 64-bit. Nothing worked.

After a while I noticed that it was consistently at night. I could boot-up the rPi in the early morning or just before sleep, in the morning it would be disconnected again.

Then I finally had the insight to remember that the router is rebooting:

*Tested:* HomeAssistant is working and I enter the router control panel and tapped the Reboot icon. The router reboots my internet and Chromecast and other rPi are all back online after a couple minutes, but the Home Assistant network is out.
*Workaround:* Disabled the router auto-reboot function and HomeAssistant have been working for few weeks without issues. It's a valid workaround but there's clearly some issue on HomeAssistant OS on the raspberryPi that does not affect the Raspberry Pi OS.

### What operating system image do you use?

rpi3-64 (Raspberry Pi 3 64-bit OS)

### What version of Home Assistant Operating System is installed?

7.6

### Did you upgrade the Operating System.

No

### Steps to reproduce the issue

1. Connect the HomeAssistant OS rPi to a router with software reboot functionality.
2. Enter the router control panel
3. Reboot the router

### Anything in the Supervisor logs that might be useful for us?

```txt
Cannot access the logs because network is down
```

### Anything in the Host logs that might be useful for us?

```txt
Cannot access the logs because network is down
```

## System Health

version | core-2022.5.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.9
os_name | Linux
os_version | 5.10.103-v8
arch | aarch64

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 7.6
-- | --
update_channel | stable
supervisor_version | supervisor-2022.05.1
docker_version | 20.10.9
disk_total | 58.7 GB
disk_used | 4.4 GB
healthy | true
supported | true
board | rpi3-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.3.3)

</details>

<details><summary>Dashboards</summary>

dashboards | 1
-- | --
resources | 0
mode | auto-gen

</details>

**Comment by agners** [comment]
This could be caused by a stale `NetworkManager` configuration getting activated. Can you install the "SSH & Web Terminal" add-on from the "Home Assistant Community Add-ons" section and run the following command:

```
nmcli connection show
```

If `Home Assistant OS default` is present and not active, try:

```
nmcli connection delete "Home Assistant OS default"
```

and see if that fixes  the problem.
